mubench.pipeline/tasks/implementations/collect_versions.py

```diff
@@ -1,17 +1,27 @@
 from data.project import Project
+from data.project_version import ProjectVersion
 from utils.data_entity_lists import DataEntityLists
 
 
 class CollectVersionsTask:
+    def __init__(self, development_mode: bool):
+        self._filter_non_compilable_versions = not development_mode
+
     def run(self, project: Project, data_entity_lists: DataEntityLists):
         return [version for version in project.versions
-                if not self.__is_filtered(project.id, version.id, data_entity_lists)]
+                if not self.__is_filtered(project.id, version, data_entity_lists, self._filter_non_compilable_versions)]
```

Comment:
Why don't we make the private `__is_filtered()` an instance method and access the field directly, instead of passing it around?